### ISSUE TITLE
Write documentation (How to use tutorial)

### ISSUE THREAD

**Issue by denibertovic** [denibertovic/docker-hs#5]
We needz docs.

**Comment by denibertovic** [comment]
Related to #12

**Comment by denibertovic** [comment]
For now there's some code in `examples/Example.hs`.

**Comment by SKoschnicke** [comment]
I would like to help.

I saw there is a small tutorial in `src/Docker/Client.hs` but it would be nice to have something like that in the `README.md` which is shown here on GitHub and at Hackage.

A more complete cookbook-style document would also be nice.

What do you think?

**Comment by jprider63** [comment]
It'd be great to have more documentation. Any effort on this front would be appreciated. 

@denibertovic, do you have any thoughts on how this should be structured? Maybe the README.md should link to src/Docker/Client.hs (or wherever it ends up), so that it's easier to find.

**Comment by denibertovic** [comment]
@SKoschnicke Yes! Any documentation effort is very welcome! Especially cookbook style docs.

It wold be nice if these things got included in the README but I'm vary of duplicating examples in haddocks and in the README. Also I'd very much like it if the examples got compiled with the projects so that we know that they work and don't get out of sync (perhaps something like a Tutorial modules like in the pipes package). 
There is an `examples` directory but it doesn't get compiled with the project currently (I usually manually just load it into ghci)...that's the source of the examples in `Client.hs` BTW.

In any case, if we could automate that inclusion of (compiled) code examples into the README that would be great. Or have the readme just point to the specific lines in the compiled examples file. cc @jprider63
